lib/libzfs/libzfs_diff.c

```diff
@@ -597,7 +597,7 @@ get_snapshot_names(differ_info_t *di, const char *fromsnap,
 	tsnlen = strlen(tosnap) - tdslen;	/* includes @ sign */
 
 	if (fsnlen <= 1 || tsnlen == 1 || (fdslen == 0 && tdslen == 0) ||
-	    (fsnlen == 0 && tsnlen == 0)) {
+	    (fdslen == 0 && tsnlen == 0)) {
```

Comment:
@behlendorf  thanks for you guidance, the resaon of i change `(fsnlen == 0 && tsnlen == 0)` to `(fdslen == 0 && tsnlen == 0)` is learn from the  comments on snapshot args format can accept,  so i think this format   `@snap1  dataset` should be invalid args format, but now after test this args format which can work correctly.